src/patterns/ec2-app.ts

```diff
@@ -203,12 +192,24 @@ export class GuEc2App {
 
     const listener = new GuHttpsApplicationListener(scope, "Listener", {
       app,
-      open: props.access.scope === AccessScope.PUBLIC, // this is true as default, which adds an ingress rule to allow all inbound traffic
+      // When open=true, AWS will create a security group which allows all inbound traffic over HTTPS
+      open: props.access.scope === AccessScope.PUBLIC,
       loadBalancer: loadBalancer,
       certificate: certificate,
       targetGroup: targetGroup,
     });
 
+    // Since AWS won't create a security group automatically when open=false, we need to add our own
+    if (props.access.scope === AccessScope.RESTRICTED) {
+      listener.connections.addSecurityGroup(
```

Comment:
This is AWS's [recommended](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-elasticloadbalancingv2.ApplicationListener.html#open) way to set more restrictive rules:

> If you want to be more selective about who can access this load balancer, set this [open] to false and use the listener's connections object to selectively grant access to the listener.

Reply:
All of the associated unit tests still pass, so I guess it must work!